### PR TITLE
26 add coordinates to navpoint entity

### PR DESCRIPTION
**Pull request by yusufzartit**

### CODE REVIEW

File: src/main/java/com/iotiq/application/domain/NavPoint.java [hunk 1]
```diff
@@ -13,7 +13,8 @@
 @Setter
 @Getter
 public class NavPoint extends AbstractPersistable<UUID> {
-    private String mapId;
+    private double latitude;
```
Comment (inciotiq) [inline src/main/java/com/iotiq/application/domain/NavPoint.java:16]: See Mesut's comments on the other pull request about adding coordinates. I think we can apply those here, too.

**Review by inciotiq — CHANGES_REQUESTED** [review]

File: src/main/java/com/iotiq/application/service/NavPointService.java [hunk 2]
```diff
@@ -53,7 +53,8 @@ public void create(NavPointCreateRequest request) {
         }
 
         NavPoint navPoint = new NavPoint();
-        navPoint.setMapId(request.mapId());
+        navPoint.setLatitude(request.latitude());
```
Comment (mesutiotiq) [inline src/main/java/com/iotiq/application/service/NavPointService.java:56]: In my opinion, we should create LocationRequest and embed to parent requests (like NavPointCreateRequest).

File: src/main/java/com/iotiq/application/messages/ExhibitionItemCreateRequest.java [hunk 3]
```diff
@@ -1,4 +1,6 @@
 package com.iotiq.application.messages;
```
Comment (mesutiotiq) [inline src/main/java/com/iotiq/application/messages/ExhibitionItemCreateRequest.java:1]: I think exhibition messages should be moved to exhibition folder like "navpoint"

Comment (yusufzartit) [reply src/main/java/com/iotiq/application/messages/ExhibitionItemCreateRequest.java:1]: will fix it in another branch.

File: src/main/java/com/iotiq/application/controller/NavPointController.java [hunk 4]
```diff
@@ -24,10 +24,10 @@ public class NavPointController {
     private final NavPointService navPointService;
 
     @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
     @PreAuthorize("hasAuthority(@NavPointAuth.CREATE)")
-    public ResponseEntity<Void> create(@RequestBody @Valid NavPointCreateRequest navPointCreateRequest) {
+    public void create(@RequestBody @Valid NavPointCreateRequest navPointCreateRequest) {
```
Comment (inciotiq) [inline src/main/java/com/iotiq/application/controller/NavPointController.java:29]: Can you remove ResponseEntity from other endpoints like you did here?

File: src/main/java/com/iotiq/application/messages/navpoint/NavPointCreateRequest.java [hunk 5]
```diff
@@ -1,11 +1,12 @@
 package com.iotiq.application.messages.navpoint;
 
+import com.iotiq.application.messages.location.LocationDto;
 import jakarta.validation.constraints.NotNull;
 
 import java.util.List;
 import java.util.UUID;
 
-public record NavPointCreateRequest(String mapId,
+public record NavPointCreateRequest(LocationDto location,
```
Comment (inciotiq) [inline src/main/java/com/iotiq/application/messages/navpoint/NavPointCreateRequest.java:9]: Can you use a distinct class called LocationRequest in requests?

**Review by inciotiq — CHANGES_REQUESTED** [review]

**Review by inciotiq — APPROVED** [review]

**Pull request merged**